static/js/src/ejsImageGallery.js

```diff
@@ -102,6 +128,35 @@ class ImageGallery {
             });
     }
 
+    _parseVideoId(url) {
+        return new Promise((resolve, _) => {
+            let urlParts;
+            let urlSubParts;
+            let id;
+
+            switch (true) {
+                //eg http://youtu.be/dQw4w9WgXcQ?feature=youtube_gdata_player
+                case (url.indexOf('?' > -1) && url.indexOf('v=') === -1):
+                    urlParts = url.split('?')
+                    urlSubParts = urlParts[0].split('/')
+                    id = urlSubParts[urlSubParts.length - 1]
+                    resolve(id)
+                    break;
+
+                // eg http://youtu.be/-wtIMTCHWuI
+                case (url.indexOf('?') === -1 && url.indexOf('v=') === -1):
+                    urlParts = url.split('/')
+                    id = urlParts[urlParts.length - 1]
+                    resolve(id)
+                    break;
+
+                // eg http://www.youtube.com/watch?v=-wtIMTCHWuI
+                default:
+                    resolve(url.split('v=')[1])
```

Comment:
Very complicated.

Can be done simpler using the built-in anchor element.

```javascript
var urlString = 'http://youtu.be/dQw4w9WgXcQ?feature=youtube_gdata_player'

const a = document.createElement('a')
a.href = urlString

console.log(a.hostname, a.pathname, a.search)

if (a.hostname === 'youtu.be') {
  console.log('Video ID:', a.pathname.substr(1, a.pathname.length))
  return
}

// otherwise proceed with parsing value from query string param 'feature='
```